### ISSUE TITLE
Add IC2 Patcher to fix IC2?

### ISSUE THREAD

**Issue by Shijikori** [EnigmaticaModpacks/Enigmatica2Expert#1983]
### URL

https://www.curseforge.com/minecraft/mc-mods/ic2-patcher

### Why would you like the mod added?

I faced the issue where the QuantumSuit in IC2 can't have it's JetPack disabled. I also found out that this issue, along quite a few others, don't seem like they will be officially fixed by the IC2 developers. I've found this mod which claims to patch most of these issues. I haven't tried it myself yet but I intend to give it a shot as not being able to disable the JetPack is very annoying.

**Comment by Shijikori** [comment]
I have tried IC2 Patcher and it does fix issues with keybinds.

**Comment by ElektroKill** [comment]
This would require updating IC2 ti the latest beta version. Not sure what exactly changed in the versions between the current one and the latest one but there could be breaking changes!

**Comment by NielsPilgaard** [comment]
Last time I updated IC2 it introduced several dupe bugs, so I'm reluctant to do that 😅

**Comment by ReiDaTecnologia** [comment]
God damnit xD
but ngl, the jetpack bug is really annoying...

**Comment by ElektroKill** [comment]
> Last time I updated IC2 it introduced several dupe bugs, so I'm reluctant to do that 😅

Yeah, IC2 is well known for dupe bugs... I think an update would be useful as it fixes some bugs within IC2 and allows usage of the mod which fixes even more bugs and annoyances. The IC2 Patcher mod also has a better fix for the Batch Crafter duplication bug instead of just disabling it outright allowing people to use the block in their worlds instead of having it disabled.

IC2Patcher has such a strict dependency on the version because it uses binary patches to patch the bugs in IC2 which easily break when the code is not the same.

**Comment by RaVenInTheDark** [comment]
So, the only reason we can't do this is because of the dependency? Does anyone know if this fixes the issues with the enet failing sometimes?

**Comment by NielsPilgaard** [comment]
If it does then I'm less reluctant to update.

**Comment by Shijikori** [comment]
> So, the only reason we can't do this is because of the dependency? Does anyone know if this fixes the issues with the enet failing sometimes?

The bugs IC2 Patcher patches are listed on the CurseForge page. No mentions of enet. I took a look at their issue tracker on their GitHub, I didn't find a specific reference to enet. You can request a bug to be patched on there.

**Comment by ThePixelbrain** [comment]
As of recently, IC2Patcher has a patch for an enet bug. This might be something worth revisiting.

**Comment by Shijikori** [comment]
Yep, they merged an enet related patch in may : [https://github.com/Su5eD/IC2-Patcher/pull/9](https://github.com/Su5eD/IC2-Patcher/pull/9)

**Comment by Shijikori** [comment]
Good job on that btw @ThePixelbrain

**Comment by Discomanco** [comment]
This likely wont happen.

IC2Patcher needs a very recent version of IC2. Any IC2 version above the one we're using right now, have autocrafting with AE2 completely broken. It works with AE2UEL, but that's not a switch we can do outside of an update that would require people to remake worlds, which is not ideal, so that also wont happen.

We tried for E2E 1.90, but the IC2 issues with AE2 were not acceptable, and it wont be fixed from either side.

**Comment by Kanzaji** [comment]
👀
https://github.com/Su5eD/IC2-Patcher/pull/12

**Comment by MommyHeather** [comment]
> 👀 [Su5eD/IC2-Patcher#12](https://github.com/Su5eD/IC2-Patcher/pull/12)
Fucking madlad made a pr just so we could get the mod in e2e

**Comment by Kanzaji** [comment]
Maybe we will open this up again 😅?
https://legacy.curseforge.com/minecraft/mc-mods/ic2-patcher/files/4777220
